kernel/comps/block/src/device_id.rs

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::collections::btree_set::BTreeSet;
+
+use device_id::{DeviceId, MajorId, MinorId};
+use id_alloc::IdAlloc;
+use ostd::sync::Mutex;
+use spin::Once;
+
+use crate::Error;
+
+/// The upper limit for block device major IDs (not included).
+const MAJOR_MAX: u16 = 512;
```

Comment:
Add a reference to the Linux source code or documentation for this magic value.

In addition, usually a maximum value is inclusive. For example, `u32::max()` returns the maximum value that is within the valid range. So reduce the value by 1.

In addition, this upper bound may be interesting to the user. So it should be public.

```
/// The maximum value of the major device ID of a block device.
pub const MAX_MAJOR: u16 = 511;
```